gdal/frmts/rmf/rmflzw.cpp

```diff
@@ -85,8 +91,8 @@ static void LZWUpdateTab(LZWStringTab *poCodeTab, GUInt32 iPred, char bFoll)
 /* It will NOT notice if the table is full. This must be handled        */
 /* elsewhere                                                            */
 /* -------------------------------------------------------------------- */
-    GUInt32 nLocal = (iPred + bFoll) | 0x0800;
-    nLocal = (nLocal*nLocal >> 6) & 0x0FFF;      // middle 12 bits of result
+    GUInt32 nLocal = CPLUnsanitizedAdd<GUInt32>(iPred, bFoll) | 0x0800;
+    nLocal = (UnsanitizedMul(nLocal, nLocal) >> 6) & 0x0FFF;      // middle 12 bits of result
```

Comment:
Don't know if it's better, but

```
(nLocal * nLocal) >> 6
```

is the same as

```
(nLocal >> 3) * (nLocal >> 3)
```

Reply:
Perhaps. This code is not mine, so I didn't want to mess up too much with that. Doing as you suggest is very slightly less efficient, and doesn't necessarily guarantee that an overflow won't occurr. I'll let this as it, as overflow in this sort of code is more or less intended I think